tokio-sync/src/mpsc/bounded.rs

```diff
@@ -219,6 +219,24 @@ impl<T> TrySendError<T> {
     pub fn into_inner(self) -> T {
         self.value
     }
+
+    /// Did the send fail because the channel has been closed?
+    pub fn was_closed(&self) -> bool {
+        if let ErrorKind::Closed = self.kind {
+            true
+        } else {
+            false
+        }
+    }
+
+    /// Did the send fail because the channel was at capacity?
+    pub fn was_full(&self) -> bool {
```

Comment:
A really minor nit, but these methods in `crossbeam-channel` are called `is_full` and `is_disconnected`. I don't have strong preferences, but it'd be nice to be consistent in naming. :)

https://docs.rs/crossbeam-channel/0.3.6/crossbeam_channel/enum.TrySendError.html#method.is_full

Reply:
Interesting. I explicitly chose to name them in the past tense because you don't know whether the channel _is_ full, just whether it _was_ full. I don't feel particularly strongly either way though. One could also argue that the error "is" full/closed. that just struck me as an odd statement to make, since the error _itself_ isn't full or closed, it just represents the state of the channel.

Reply:
It's not that `is_full` means that the channel is (currently) full, it means the error is an "is full error".